### PR TITLE
KEYCLOAK-12615 HS384 and HS512 support for Client Authentication by Client Secret Signed JWT

### PR DESCRIPTION
**Pull request by tnorimat**
This document PR is for https://github.com/keycloak/keycloak/pull/6633 .

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@tnorimat Thanks, approving.

@andymunro Do you have a chance to review as well please? Thanks

File: securing_apps/topics/oidc/java/client-authentication.adoc [hunk 1]
```diff
@@ -66,11 +66,16 @@ The client has a secret, which needs to be known to both the adapter (applicatio
...
 "credentials": {
   "secret-jwt": {
-    "secret": "19666a4f-32dd-4049-b082-684c74115f28"
+    "secret": "19666a4f-32dd-4049-b082-684c74115f28",
+    "algorithm": "HS512"
   }
 }
 ----
 
+"algorithm" field specifies the algorithm for Signed JWT using Client Secret. It needs to be the one of the following values : HS256, HS384, and HS512. As for the details about them, please refer to https://tools.ietf.org/html/rfc7518#section-3.2[JSON Web Algorithms (JWA)].
```
Comment (andymunro) [inline securing_apps/topics/oidc/java/client-authentication.adoc]: remove "the" before "one"

Change "As for the details about them" to "For details"

File: securing_apps/topics/oidc/java/client-authentication.adoc [hunk 2]
```diff
@@ -66,11 +66,16 @@ The client has a secret, which needs to be known to both the adapter (applicatio
...
+
+This "algorithm" field is optional so that HS256 is applied automatically if "algorithm" field does not exist on `keycloak.json` file.
```
Comment (andymunro) [inline securing_apps/topics/oidc/java/client-authentication.adoc]: add "the" before "algorithm"
add "the" before 'keycloak.json'

**Review by andymunro — DISMISSED** [review]
approved with a few changes suggested.

File: securing_apps/topics/oidc/java/client-authentication.adoc (hunk 1 above)
Comment (tnorimat) [reply securing_apps/topics/oidc/java/client-authentication.adoc]: Yes, I'll fix it as you said.

File: securing_apps/topics/oidc/java/client-authentication.adoc (hunk 2 above)
Comment (tnorimat) [reply securing_apps/topics/oidc/java/client-authentication.adoc]: I see. I'll fix it as you said.

**Comment by tnorimat** [comment]
@andymunro  Thank you for your review. I've fixed the points that you've commented on.

**Review by mposolda — APPROVED** [review]
@andymunro I've approved this. Do you have a chance to re-review please? Thanks

**Pull request merged**